prow/plank/controller.go

```diff
@@ -135,6 +135,25 @@ func (c *Controller) canExecuteConcurrently(pj *prowapi.ProwJob) bool {
 		c.log.WithFields(pjutil.ProwJobFields(pj)).Debugf("Not starting another instance of %s, already %d running.", pj.Spec.Job, numPending)
 		return false
 	}
+
+	var olderMatchingPJs []prowapi.ProwJob
+	for _, foundPJ := range c.pjs {
+		if foundPJ.Status.State != prowapi.TriggeredState {
+			continue
+		}
+		if foundPJ.Spec.Job != pj.Spec.Job {
+			continue
+		}
+		if foundPJ.CreationTimestamp.Before(&pj.CreationTimestamp) {
+			olderMatchingPJs = append(olderMatchingPJs, foundPJ)
```

Comment:
nit: Only the length of this array is ever consumed so it could just be an integer.